pkg/apis/certmanager/v1alpha1/types.go

```diff
@@ -139,6 +140,26 @@ type ACMEIssuerDNS01ProviderRoute53 struct {
 	Region          string            `json:"region"`
 }
 
+// TSGIG HMAC hashing codes
+type TSIGAlgorithm string
+
+const (
+	HmacMD5    TSIGAlgorithm = "HmacMD5"
+	HmacSHA1   TSIGAlgorithm = "HmacSHA1"
+	HmacSHA256 TSIGAlgorithm = "HmacSHA256"
+	HmacSHA512 TSIGAlgorithm = "HmacSHA512"
+)
+
+// ACMEIssuerDNS01ProviderRFC2136 is a structure containing the DNS
+// configuration for RFC2136
+type ACMEIssuerDNS01ProviderRFC2136 struct {
+	Nameserver    string            `json:"nameserver"`
+	TSIGSecret    SecretKeySelector `json:"tsigSecretSecretRef"`
+	TSIGKey       string            `json:"tsigKey"`
+	TSIGAlgorithm TSIGAlgorithm     `json:"tsigAlgorithm"`
+	Timeout       string            `json:"timeout"`
```

Comment:
The 'timeout' field isn't actually used within the provider, in favour of a static global timeout.

None of the other DNS providers have this field, so for now I think we're best to remove it from our API surface. In future, we may add it back in (potentially as a top-level timeout field on the issuer).

Corresponding docs need updating too to reflect this!